### PR TITLE
Tw 618/list nft

### PR DESCRIPTION
**Pull request by Boltman92**

### CODE REVIEW

**Review by svjatoslavtt — APPROVED** [review]

File: src/screens/collection/components/collectible-item.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+  useEffect(() => setWidth(layoutWidth), [layoutWidth]);
+
+  useEffect(() => {
+    tezos.contract
+      .at<ObjktContractInterface | FxHashContractInterface>(item.highestOffer?.marketplace_contract ?? OBJKT_CONTRACT)
+      .then(setOffer);
+  }, []);
+
+  const lastPrice =
```
Comment (svjatoslavtt) [inline src/screens/collection/components/collectible-item.tsx:57]: I think it is hard to understand. Can we rewrite something like this
```
const lastPrice  = useMemo(() => {
    const lastPrice = formatAssetAmount(mutezToTz(BigNumber(item.lastPrice?.price), item.lastPrice?.decimals));
    
    if (isDefined(item.lastPrice) && isDefined(item.lastPrice.price)) {
       return `${price} ${item.symbol}`;
    }
    
    return '---';
}, [item]);
```

File: src/screens/collection/components/collectible-item.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  const navigateToObjktForBuy = `https://objkt.com/asset/${collectionContract}/${item.id}`;
+
+  const handlePress = async () => {
+    if (!isHolder && isListed) {
+      return openUrl(navigateToObjktForBuy);
+    }
+
+    const getTransferParams = () => {
+      if (isDefined(offer)) {
+        if ('fulfill_offer' in offer?.methods) {
```
Comment (svjatoslavtt) [inline src/screens/collection/components/collectible-item.tsx:87]: mb in the future we must create some enum or something like this for over the methods of objkt and hxhash

**Review by svjatoslavtt — APPROVED** [review]

File: src/utils/image.utils.ts [hunk 3]
```diff
@@ -57,12 +57,19 @@ export const formatCollectibleObjktMediumUri = (assetSlug: string) => {
...
-  return `${OBJKT_ORIGIN}/${OBJKT_RESIZE_3}/${
-    artifactUri.includes('ipfs://') ? artifactUri.substring(IPFS_PROTOCOL_PREFIX.length) : artifactUri
-  }/artifact`;
+  if (artifactUri.includes('fxhash')) {
+    const [artifactUriWithoutFxHash, fxhash] = artifactUri.split('?');
+
+    return `${OBJKT_ORIGIN}/${OBJKT_RESIZE_3}/${checkIpfs(artifactUriWithoutFxHash)}/artifact/index.html?${fxhash}`;
+  }
+
+  return `${OBJKT_ORIGIN}/${OBJKT_RESIZE_3}/${checkIpfs(artifactUri)}/artifact`;
```
[the remark this answers is not in the document]
Comment (Boltman92) [reply src/utils/image.utils.ts]: are you sure that regEx would be better to understand?

File: src/screens/collection/collection.tsx [hunk 4]
```diff
@@ -65,6 +74,8 @@ export const Collection = () => {
     </View>
   );
 
+  const keyExtractor = useCallback(item => `${item.address}_${item.id}`, []);
```
Comment (svjatoslavtt) [inline src/screens/collection/collection.tsx:77]: you can just move it above the component and don't use `useCallback` that use some resources too

**Review by svjatoslavtt — APPROVED** [review]

**Review by IrynaKhyzhynska — APPROVED** [review]
android done 1.15_746

**Pull request merged**